contract/wizards/contract_manually_create_invoice.py

```diff
@@ -52,7 +59,24 @@ def create_invoice(self):
         self.ensure_one()
         invoices = self.env["account.move"]
         for contract in self.contract_to_invoice_ids:
-            invoices |= contract.recurring_create_invoice()
+            try:
+                invoices |= contract.recurring_create_invoice()
+            except (
+                AccessDenied,
+                AccessError,
+                MissingError,
+                UserError,
+                ValidationError,
+            ) as oe:
+                raise UserError(
+                    _(
+                        "Failed to process the contract %(name)s [id: %(id)s]:\n%(ue)s",
+                        name=contract.name,
+                        id=contract.id,
+                        ue=repr(oe),
+                    )
+                ) from oe
```

Comment:
From v14, only this line was changed to comply with pre-commit/flake8 requirement:

```
contract/wizards/contract_manually_create_invoice.py:71:17: B904 Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling.  See https://docs.python.org/3/tutorial/errors.html#exception-chaining for details.
```